freqtrade/wallets.py

```diff
@@ -0,0 +1,38 @@
+# pragma pylint: disable=W0603
+""" Wallet """
+import logging
+from typing import Dict, Any
+from collections import namedtuple
+from freqtrade.exchange import Exchange
+
+logger = logging.getLogger(__name__)
+
+
+class Wallets(object):
+
+    # wallet data structure
+    wallet = namedtuple(
+        'wallet',
+        ['exchange', 'currency', 'free', 'used', 'total']
+    )
+
+    def __init__(self, exchange: Exchange) -> None:
+        self.exchange = exchange
+        self.wallets: Dict[str, Any] = {}
+        self.update()
+
+    def update(self) -> None:
+        balances = self.exchange.get_balances()
+
+        for currency in balances:
+            info = {
+                'exchange': self.exchange.id,
+                'currency': currency,
+                'free': balances[currency]['free'],
+                'used': balances[currency]['used'],
+                'total': balances[currency]['total']
+            }
+
+            self.wallets[currency] = self.wallet(**info)
```

Comment:
this seems slightly complicated ...
why not init the named-tuple directly without the intermediate step of a dict?

also - personally i would subclass namedtuple here to get the benefit of typehints (see `arguments.py` - `TimeRange` for a sample ...). mypy can then help detect problematic usages of this class ...

Reply:
I know ! stupid mypy gives this error if I init directly: https://github.com/python/mypy/issues/2996